### PR TITLE
Improved BackwardSlicer robustness

### PR DESCRIPTION
**Pull request by uxmal**
This PR should improve the decompilation results of the binary provided in issue #691.

### CODE REVIEW

**Comment by uxmal** [comment]
Please chime in with comments on this PR, @ptomin .

File: src/Decompiler/Scanning/BackwardSlicer.cs [hunk 1]
```diff
@@ -685,29 +716,60 @@ public SlicerResult VisitBinaryExpression(BinaryExpression binExp, BackwardSlice
...
+                //if (seFound != null)
+                //    return seFound;
+                //foreach (var live in this.Live)
+                //{
+                //    if (live.Value.Type != ContextType.Jumptable)
+                //        continue;
+                //    seFound = TryFoundAllConditions(binExp, domLeft, live.Key, live.Value);
+                //    if (seFound != null)
+                //        return seFound;
+                //}
```
Comment (ptomin) [inline src/Decompiler/Scanning/BackwardSlicer.cs]: Commented code should be removed

File: src/Decompiler/Scanning/BackwardSlicer.cs [hunk 2]
```diff
@@ -733,6 +795,39 @@ public SlicerResult VisitBinaryExpression(BinaryExpression binExp, BackwardSlice
...
 
+        /// <summary>
+        /// If all necessary conditions for a bounded indirect jump have been met, we can
+        /// set the jump table bounds and proceed.
+        /// </summary>
+        private SlicerResult TryFoundAllConditions(
+            BinaryExpression subtraction, 
+            StorageDomain domLeft, 
+            Expression indexCandidate, 
+            BackwardSlicerContext ctx)
```
Comment (ptomin) [inline src/Decompiler/Scanning/BackwardSlicer.cs]: Is it unused method? If so should be removed

File: src/Decompiler/Scanning/BackwardSlicer.cs (hunk 1 above)
Comment (uxmal) [reply src/Decompiler/Scanning/BackwardSlicer.cs]: Removed

File: src/Decompiler/Scanning/BackwardSlicer.cs (hunk 2 above)
Comment (uxmal) [reply src/Decompiler/Scanning/BackwardSlicer.cs]: Removed

**Pull request merged**